### ISSUE TITLE
How to add options route to the controller

### ISSUE THREAD

**Issue by trailblazer2006** [TypedProject/ts-express-decorators#639]
## Information

- **Version:** 5.x
- **Type:** Issue/Story

## Description
I want to add an options route to my controller. I can't find options decorator in tsed. How could I solve this problem

## Old code
```
export default (apiRouter: Router) => {
  apiRouter.get('/dashboards/:id', api.getDocument);
  apiRouter.get('/dashboards', api.getDocuments);
  apiRouter.post('/dashboards', api.createDocument);
  apiRouter.patch('/dashboards/:id', api.updateDocument);
  apiRouter.delete('/dashboards/:id', api.deleteDocument);
  apiRouter.options('/dashboards/*', api.options);
};
```

## New code
```typescript
import {Controller, Delete, Get, Patch, Post, Status} from '@tsed/common';

@Controller('/dashboards')
export class DashboardsController {
  @Get('/:id')
  async getDocument(req): Promise<any> {
    return await api.getDocument(req.params.id, req.query);
  }
  @Get()
  async getDocuments(req): Promise<any> {
    return await api.getDocuments(req.query, req.query.lastAccessed === 'true');
  }
  @Status(201)
  @Post()
  async createDocument(req): Promise<any> {
    return await api.createDocument(req.body, req.query);
  }
  @Status(201)
  @Patch('/:id')
  async updateDocument(req): Promise<any> {
    return  await api.updateDocument({
      documentId: req.params.id,
      body: req.body,
      query: req.query,
    });
  }
  @Status(204)
  @Delete('/:id')
  async deleteDocument(req: GatewayRequest): Promise<void> {
    await api.deleteDocument(req.params.id);
  }
  // @Options('/*')
  // async options(req, res): Promise<void> {
  //   const resp = await api.getOptions();
  //   return res
  //     .set('Allow', resp.headers.get('Allow') || undefined)
  //     .sendStatus(200)
  //     .end();
  // }
}

```

**Comment by Romakita** [comment]
Hello @trailblazer2006 ,
You've right, this decorator is missing in the current version. I just added this decorator in v5.31.0 (release will be here soon, in 15min).

See you
Romain

**Comment by Romakita** [comment]
Options will work like Get or Post
```typescript
@Options('/*')
async options(@Res() res: Res): Promise<void> {
  res.set('Allow', resp.headers.get('Allow') || undefined)
  
  return await api.getOptions();
}

**Comment by Romakita** [comment]
Released with v5.31.0 https://tsed.io/docs/controllers.html#routing

**Comment by trailblazer2006** [comment]
Thank you
